### DISCUSSION TITLE
RTSP / HLS stream is crashing due to buffer issue

### DISCUSSION THREAD

**Discussion by jjeong95** [bluenviron/mediamtx#3864] [Questions]
### Question

Hello,

I'm currently facing a bit of weird issue with using Media MTX to convert an RTSP to HLS.

Currently, source of my stream comes from a Ring camera where we get its RTSP stream through a platform called `Scrypted` as Ring does not provide an RTSP out of the box. With the RTSP from Scrypted, I've used the the FFmpeg command to publish an HLS stream on the VM that has Media MTX running.

FFmpeg command `ffmpeg -rtsp_transport tcp -re -stream_loop -1 -i rtsp://ipaddress/e8f6788872c63846 -c copy -f rtsp rtsp://ipaddress:8554/mystream1`

However, everytime I have the HLS running it crashes constantly with the following logs below.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 1932096, current: 1927360; changing to 1932097. This may result in incorrect timestamps in the output file.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 1932097, current: 1928384; changing to 1932098. This may result in incorrect timestamps in the output file.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 1932098, current: 1929408; changing to 1932099. This may result in incorrect timestamps in the output file.

[vist#0:1/h264 @ 0x64fbda007980] corrupt decoded frame

[aac @ 0x64fbda00cec0] Queue input is backward in time

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 1940288, current: 1935552; changing to 1940289. This may result in incorrect timestamps in the output file.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 1940289, current: 1936576; changing to 1940290. This may result in incorrect timestamps in the output file.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 1940290, current: 1937600; changing to 1940291. This may result in incorrect timestamps in the output file.

[aac @ 0x64fbda00cec0] Queue input is backward in timeitrate=N/A dup=8 drop=0 speed=1.01x    

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 2801472, current: 2796736; changing to 2801473. This may result in incorrect timestamps in the output file.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 2801473, current: 2797760; changing to 2801474. This may result in incorrect timestamps in the output file.

[rtsp @ 0x64fbda015dc0] Non-monotonic DTS in output stream 0:1; previous: 2801474, current: 2798784; changing to 2801475. This may result in incorrect timestamps in the output file.

I've tried to verify whether the RTSP stream from Scrypted or the output for Media MTX is having issue, but when I was running the RTSP from either of them on VLC, none of them seemed to have any crashes and the stream ran continuously without an issue.

Just to rule out that RTSP is not an issue, I've also tried running a publicly available RTSP stream from https://rtsp.stream/ and use the same FFmpeg command above to public an HLS stream and that stream does not seem to have a DTS output issue ( or buffer issue of some sort ) shown above.

What I'm not able to rule out is, if Ring RTSP or RTSP from Scrypted is not compatible with Media MTX.

If anyone could share some sort of insight that would be much appreciated and if you require any more additional informations please do ask away.

**Comment by aler9** [comment]
Hello, log lines like "Non-monotonic DTS in output stream" are generated by FFmpeg, not by the server, before the stream even reaches the server.
This means that there's something wrong in the original stream.
However, FFmpeg provides utilities to fix timestamps (DTS / PTS) and to re-encode the stream, there are plenty of instructions available online, even though they might not be enough, depending on the content on the original stream.
